c-pallets/file-bank/src/lib.rs

```diff
@@ -424,12 +432,12 @@ pub mod pallet {
 		/// - `file_hash`: The beneficiary related to signer account.
 		/// - `file_size`: File size calculated by consensus.
 		/// - `slice_info`: List of file slice information.
-		/// - `user`: The first user to upload files.
+		/// - `user`: The first user to upload files. // Does it mean origin? I don't see the parameter
 		#[transactional]
-		#[pallet::weight(<T as pallet::Config>::WeightInfo::upload())]
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::upload())] // should depend on slice_info value
 		pub fn upload(
 			origin: OriginFor<T>,
-			file_hash: Vec<u8>,
+			file_hash: Vec<u8>, // Hashes have a fixed size, Vec<u8> doesn't seem the best choice
```

Comment:
I define a type for it:
https://github.com/CESSProject/cess/pull/88/files#diff-9c37b1a61a6e9394aac1e9383bf9d3fac9539378d1af8fcd75aed4f8a4f42967R458
defined here:
https://github.com/CESSProject/cess/pull/88/files#diff-1b48106f9cdd51de414e128717cc6b39dc2059cc0f9087f3c456ce63c2e4c6d1R12